File: sources/Dirket/us/dev/api/interfaces/Toggleable.java
package us.dev.api.interfaces;

/**
 * Created by Foundry on 11/15/2015.
 */
public interface Toggleable {
    default void toggle() {
        this.setRunning(!this.isRunning());
    }

    boolean isRunning();

    void setRunning(boolean running);

    void onEnable();

    void onDisable();
}

File: sources/Leaf devEnv/com/leafclient/leaf/management/utils/Toggleable.kt
package com.leafclient.leaf.management.utils

interface Toggleable {

    /**
     * Returns whether this toggleable object is running or not
     */
    var isRunning: Boolean

    /**
     * Method executed when the toggleable is enabled
     */
    fun onEnable() {}

    /**
     * Method executed when the toggleable is disabled
     */
    fun onDisable() {}

    /**
     * Modifies the [isRunning] to its opposite
     */
    fun toggle() {
        isRunning = !isRunning
    }

}